common/util.go

```diff
@@ -519,13 +519,16 @@ func CreateHistoryStartWorkflowRequest(
 	}
 
 	histRequest.FirstDecisionTaskBackoffSeconds = Int32Ptr(firstDecisionTaskBackoffSeconds)
-
+	expirationInSeconds := Int32Default(startRequest.ExecutionStartToCloseTimeoutSeconds)
```

Comment:
If the user doesn't specify retry expiration timeout, only the max retry attempts, assigning this value will add an additional retry condition and limit the retry to happen within the workflow timeout I think.

Reply:
I think if this isn't set then `GenerateWorkflowStartTasks` will perform the same logic so this will be a no-op in that case? since expiration timestamp will be now + Int32Default(startRequest.ExecutionStartToCloseTimeoutSeconds) and in GenerateWorkflowStartTasks we are calculating `workflowTimeoutTimestamp := startTime.Add(workflowTimeoutDuration)` which should give us the same values if retry expiration is not set